### PR TITLE
[UNR-3128] Fix startup actors receiving empty lists

### PR DESCRIPTION
**Pull request by mironec**
#### Description
Scenario where this would break:
1) A client checks out a `bNetLoadOnClient` startup actor which has a replicated TArray, with let's say [0,1,2]
2) The client walks away so the actor is no longer checked out
3) The startup actor is updated by the server, so the array is now empty []
4) The client comes back to the startup actor and checks it out again, the empty array would not be applied

#### Release note
TODO

#### Tests
Soon to come in EngineNetTest.

#### Documentation
N/A

#### Primary reviewers
@mattyoung-improbable @joshuahuburn

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDK/Private/Utils/ComponentReader.cpp [hunk 1]
```diff
@@ -108,7 +108,30 @@ void ComponentReader::ApplySchemaObject(Schema_Object* ComponentObject, UObject*
 
 	TArray<UProperty*> RepNotifies;
 
-	for (uint32 FieldId : UpdatedIds)
+	TArray<Schema_FieldId> InitialIds;
+	if (UNLIKELY(bIsInitialData))
```
Comment (mironec) [inline SpatialGDK/Source/SpatialGDK/Private/Utils/ComponentReader.cpp]: Poll: react with 👍  or 👎 if you like the unlikely macro.

Comment (samiwh) [reply SpatialGDK/Source/SpatialGDK/Private/Utils/ComponentReader.cpp]: I feel we shouldn't do things like this until we've at least tried to measure something or refactor code to not have the problem in the first place.

File: SpatialGDK/Source/SpatialGDK/Private/Utils/ComponentReader.cpp [hunk 2]
```diff
@@ -108,7 +108,28 @@ void ComponentReader::ApplySchemaObject(Schema_Object* ComponentObject, UObject*
...
+		for (int32 HandleIndex = 0; HandleIndex < BaseHandleToCmdIndex.Num(); HandleIndex++)
+		{
+			const FRepParentCmd& Parent = Parents[Cmds[BaseHandleToCmdIndex[HandleIndex].CmdIndex].ParentIndex];
+			if (GetGroupFromCondition(Parent.Condition) == SchemaType)
+			{
+				InitialIds.Add(HandleIndex + 1);
+			}
+		}
+	}
+	const TArray<Schema_FieldId>& IdsToIterate = bIsInitialData ? InitialIds : UpdatedIds;
```
Comment (joshuahuburn) [inline SpatialGDK/Source/SpatialGDK/Private/Utils/ComponentReader.cpp:130]: So the `UpdateIds` will 100% not contain something we might miss in the in `InitialIds`?

**Comment by mironec** [comment]
Superseded by https://github.com/spatialos/UnrealGDK/pull/2092

**Pull request closed without merging**